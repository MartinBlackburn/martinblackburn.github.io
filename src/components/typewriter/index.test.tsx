// libraries
import { act, render } from "@testing-library/react";

// component under test
import Typewriter from "./";

// mocks
jest.useFakeTimers();
jest.spyOn(global, "setInterval");
jest.spyOn(global, "clearInterval");

describe("Typewriter component", () => {
    beforeEach(() => {
        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.clearAllTimers();
    });

    test("renders correctly", () => {
        const { asFragment } = render(<Typewriter />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders correctly after 1.4 seconds", () => {
        const { asFragment, rerender } = render(<Typewriter />);

        act(() => {
            jest.advanceTimersByTime(1400);
        });

        expect(asFragment()).toMatchSnapshot();
    });

    test("renders correctly after 5 seconds", () => {
        const { asFragment, rerender } = render(<Typewriter />);

        act(() => {
            jest.advanceTimersByTime(5000);
        });

        expect(asFragment()).toMatchSnapshot();
    });

    test("calls setInterval on mount", () => {
        render(<Typewriter />);

        expect(setInterval).toHaveBeenCalledTimes(1);
    });

    test("calls clearInterval on unmount", () => {
        const { unmount } = render(<Typewriter />);

        unmount();

        expect(clearInterval).toHaveBeenCalledTimes(1);
    });
});
