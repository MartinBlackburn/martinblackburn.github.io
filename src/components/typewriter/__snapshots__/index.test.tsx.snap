// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typewriter component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="typewriter"
  />
</DocumentFragment>
`;

exports[`Typewriter component renders correctly after 1.4 seconds 1`] = `
<DocumentFragment>
  <div
    class="typewriter"
  />
</DocumentFragment>
`;

exports[`Typewriter component renders correctly after 5 seconds 1`] = `
<DocumentFragment>
  <div
    class="typewriter"
  />
</DocumentFragment>
`;
